### PR TITLE
Consistency fixes

### PR DESCRIPTION
**Pull request by michael-o**
As discussed on the list. Here are the points mentioned broken down to separate commits.

### CODE REVIEW

File: httpcore5/src/main/java/org/apache/hc/core5/http/impl/nio/LengthDelimitedDecoder.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public int read(final ByteBuffer dst) throws IOException {
             setCompleted();
             if (this.len < this.contentLength) {
                 throw new ConnectionClosedException(
-                                "Premature end of Content-Length delimited message body (expected: %,d; received: %,d)",
+                                "Premature end of Content-Length delimited message body (expected: %d; received: %d)",
```
Comment (garydgregory) [inline httpcore5/src/main/java/org/apache/hc/core5/http/impl/nio/LengthDelimitedDecoder.java]: I do not understand why you'd want to make the message less readable by humans.

File: httpcore5/src/main/java/org/apache/hc/core5/util/Args.java [hunk 2]
```diff
@@ -114,7 +114,7 @@ private static IllegalArgumentException illegalArgumentExceptionNotNull(final St
 
     public static <T extends CharSequence> T notEmpty(final T argument, final String name) {
         if (argument == null) {
-            throw illegalArgumentExceptionNotNull(name);
+            throw NullPointerException(name);
```
Comment (garydgregory) [inline httpcore5/src/main/java/org/apache/hc/core5/util/Args.java]: Why not use `Objects.requireNonNull()`?

File: httpcore5/src/main/java/org/apache/hc/core5/http/impl/nio/LengthDelimitedDecoder.java (hunk 1 above)
Comment (michael-o) [reply httpcore5/src/main/java/org/apache/hc/core5/http/impl/nio/LengthDelimitedDecoder.java]: As discussed, it is not consistent across locales. Thus the messages are never stable.

File: httpcore5/src/main/java/org/apache/hc/core5/util/Args.java (hunk 2 above)
Comment (michael-o) [reply httpcore5/src/main/java/org/apache/hc/core5/util/Args.java]: Sure, why not.

**Comment by ok2c** [comment]
@michael-o I am fine with everything except for 21fb5c1. I would prefer to keep `LangUtils` until 5.1.

**Comment by garydgregory** [comment]
> @michael-o I am fine with everything except for [21fb5c1](https://github.com/apache/httpcomponents-core/commit/21fb5c1fb341e224abc7cec1d913e03f4c7d012e). I would prefer to keep `LangUtils` until 5.1.

If a public API is released in 5.x, we cannot drop it until 6.0, we should drop it now.

**Comment by michael-o** [comment]
> 
> 
> > @michael-o I am fine with everything except for [21fb5c1](https://github.com/apache/httpcomponents-core/commit/21fb5c1fb341e224abc7cec1d913e03f4c7d012e). I would prefer to keep `LangUtils` until 5.1.
> 
> If a public API is released in 5.x, we cannot drop it until 6.0, we should drop it now.

I actually had the same question. I don't mind keeping it, but I see no benefit in doing so.

**Comment by michael-o** [comment]
I have moved the LangUtils removal on top to be easily moved to a separate branch if it shall be retained for any reason in 5.0.

**Comment by ok2c** [comment]
> If a public API is released in 5.x, we cannot drop it until 6.0, we should drop it now.

@michael-o We do not need to drop it. We can deprecate it and stop using it. What I dislike about `Objects` is that hideous `static int hash(Object... values)` with an extra array allocation due to vararg signature. I would like to upgrade to JRE 8 or 11 and then consider deprecating `LangUtils` in favor of `Objects`

**Comment by michael-o** [comment]
> 
> 
> > If a public API is released in 5.x, we cannot drop it until 6.0, we should drop it now.
> 
> @michael-o We do not need to drop it. We can deprecate it and stop using it. What I dislike about `Objects` is that hideous `static int hash(Object... values)` with an extra array allocation due to vararg signature. I would like to upgrade to JRE 8 or 11 and then consider deprecating `LangUtils` in favor of `Objects`

Now I understand, makes sense. How would Java 8 or 11 make it any better? Those has methods exist since Java 7.

**Comment by ok2c** [comment]
> > > If a public API is released in 5.x, we cannot drop it until 6.0, we should drop it now.
> > 
> > 
> > @michael-o We do not need to drop it. We can deprecate it and stop using it. What I dislike about `Objects` is that hideous `static int hash(Object... values)` with an extra array allocation due to vararg signature. I would like to upgrade to JRE 8 or 11 and then consider deprecating `LangUtils` in favor of `Objects`
> 
> Now I understand, makes sense. How would Java 8 or 11 make it any better? Those has methods exist since Java 7.

@michael-o I recall reading that _possibly_ some newer JVMs include a special optimization routine specifically for `Objects#hash` but it had no time to investigate it.

**Comment by michael-o** [comment]
> 
> 
> > > > If a public API is released in 5.x, we cannot drop it until 6.0, we should drop it now.
> > > 
> > > 
> > > @michael-o We do not need to drop it. We can deprecate it and stop using it. What I dislike about `Objects` is that hideous `static int hash(Object... values)` with an extra array allocation due to vararg signature. I would like to upgrade to JRE 8 or 11 and then consider deprecating `LangUtils` in favor of `Objects`
> > 
> > 
> > Now I understand, makes sense. How would Java 8 or 11 make it any better? Those has methods exist since Java 7.
> 
> @michael-o I recall reading that _possibly_ some newer JVMs include a special optimization routine specifically for `Objects#hash` but it had no time to investigate it.

Alright, will move it out and merge the rest to master.

**Comment by michael-o** [comment]
Changes merged except `LangUtils` removal.

**Pull request closed without merging**